src/EditorFeatures/Core/EditorConfigSettings/Updater/NamingStyles/SourceTextExtensions.cs

```diff
@@ -0,0 +1,186 @@
+                    newNamingStyleSection.Append("\r\n");
+                    newNamingStyleSection.Append(Section.GetHeaderTextForLanguage(language));
+                    EditorConfigFileGenerator.AppendNamingStylePreferencesToEditorConfig(rules, newNamingStyleSection, GetLanguageString(language));
+                    return WithChanges(sourceText, span, newNamingStyleSection.ToString());
+                }
+            }
+
+            return sourceText;
+
+            static SourceText WithChanges(SourceText sourceText, TextSpan span, string newText)
+            {
+                var textChange = new TextChange(span, newText);
+                return sourceText.WithChanges(textChange);
+            }
+
+            static string? GetLanguageString(Language language)
+            {
+                if (language.HasFlag(Language.CSharp) && language.HasFlag(Language.VisualBasic))
+                {
+                    return null;
+                }
+                else if (language.HasFlag(Language.CSharp))
+                {
+                    return LanguageNames.CSharp;
+                }
+                else if (language.HasFlag(Language.VisualBasic))
+                {
+                    return LanguageNames.VisualBasic;
+                }
+
+                throw new InvalidOperationException("Invalid language specified");
+            }
+        }
+
+        private static (IEnumerable<NamingRule> Common, IEnumerable<NamingRule> CSharp, IEnumerable<NamingRule> VisualBasic) GetPreferencesForAllLanguages(OptionSet options)
+        {
+            var csharpNamingStylePreferences = options.GetOption(NamingStyleOptions.NamingPreferences, LanguageNames.CSharp);
+            var vbNamingStylePreferences = options.GetOption(NamingStyleOptions.NamingPreferences, LanguageNames.VisualBasic);
+
+            var commonOptions = GetCommonOptions(csharpNamingStylePreferences, vbNamingStylePreferences);
+            var csharpOnlyOptions = GetOptionsForCSharp(csharpNamingStylePreferences, commonOptions);
+            var vbOnlyOptions = GetOptionsForVisualBasic(vbNamingStylePreferences, commonOptions);
+            return (commonOptions, csharpOnlyOptions, vbOnlyOptions);
+
+            static IEnumerable<NamingRule> GetCommonOptions(NamingStylePreferences csharp, NamingStylePreferences visualBasic)
+                => csharp.Rules.NamingRules.Intersect(visualBasic.Rules.NamingRules, NamingRuleComparerIgnoreGUIDs.Instance);
+
+            static IEnumerable<NamingRule> GetOptionsForCSharp(NamingStylePreferences csharp, IEnumerable<NamingRule> common)
+                => csharp.Rules.NamingRules.Except(common, NamingRuleComparerIgnoreGUIDs.Instance);
+
+            static IEnumerable<NamingRule> GetOptionsForVisualBasic(NamingStylePreferences visualBasic, IEnumerable<NamingRule> common)
+                => visualBasic.Rules.NamingRules.Except(common, NamingRuleComparerIgnoreGUIDs.Instance);
```

Comment:
Since both local functions have the same implementation and signature, it could be unified into just `GetOptionsForSpecificLanguage(NamingStylePreferences languagePreferences, IEnumerable<NamingRule> common)`